src/main/java/hudson/scm/browsers/VisualSVN.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * The MIT License
+ * 
+ * Copyright (c) 2016, Sun Microsystems, Inc., Kohsuke Kawaguchi, Arnost Havelka
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.scm.browsers;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.scm.EditType;
+import hudson.scm.RepositoryBrowser;
+import hudson.scm.SubversionChangeLogSet;
+import hudson.scm.SubversionChangeLogSet.Path;
+import hudson.scm.SubversionRepositoryBrowser;
+import hudson.scm.subversion.Messages;
+
+
+/**
+ * {@link RepositoryBrowser} for Subversion.
+ *
+ * https://issues.jenkins-ci.org/browse/JENKINS-30176
+ * 
+ * @author Arnost Havelka
+ * @since TODO
+ */
+public class VisualSVN extends SubversionRepositoryBrowser {
+	
+	private static final Logger LOGGER = Logger.getLogger(VisualSVN.class.getName());
+
+    @Extension
+    public static class DescriptorImpl extends Descriptor<RepositoryBrowser<?>> {
+        public String getDisplayName() {
+        	return Messages.SubversionSCM_browsers_VisualSVN();
+        }
+    }
+
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The URL of the top of the site.
+     *
+     * <p>Normalized to ends with '/', like <tt>https://demo-server.visualsvn.com/!/#tortoisesvn</tt>
+     * It may contain a query parameter like <tt>?root=foobar</tt>, so relative
+     * URL construction needs to be done with care.</p>
+     */
+    private final String url;
+
+    /**
+     * Creates a new VisualSVN object.
+     *
+     * @param                url  base URL
+     *
+     * @throws               MalformedURLException  when URL is not valid (empty)
+     */
+    @DataBoundConstructor
+    public VisualSVN(URL url) throws MalformedURLException {

```

Comment:
I am not sure if Stapler handles URLs reliably. CC @daniel-beck and @vivek

Reply:
Should make the parameter type be `String`, to match that of the getter. Otherwise this will not be compatible with Pipeline’s **Snippet Generator**, for example, and might cause mayhem in JEP-201 etc.

Reply:
Yes, it's better to fix it before merge